### PR TITLE
[Debugger] Don't Dispose() items until *after* we call RemoveItems()

### PR DESCRIPTION
**Pull request by jstedfast**
May fix https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1004450/

### CODE REVIEW

**Comment by jstedfast** [comment]
@monojenkins backport release-8.4

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeViewDataSource.cs [hunk 1]
```diff
@@ -127,41 +141,40 @@ public void Replace (ObjectValueNode node, ObjectValueNode[] replacementNodes)
 			if (index == -1)
 				return;
 
+			var removed = new List<MacObjectValueNode> ();
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeViewDataSource.cs]: Do we know how many we're removing?

Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacObjectValueTreeViewDataSource.cs]: We know how many top-level items we are removing, but that does not include the child nodes that we also need to dispose.

**Comment by sgmunn** [comment]
this is hard to validate. I've not been able to reproduce this exact crash so I cannot tell if it helps or not. I do know that this doesn't help the other crasher that we have with collapsing nodes (in tooltips).

**Comment by jstedfast** [comment]
Can this be merged?

**Comment by Therzok** [comment]
Re-triggered test run, provisioning failed.

**Comment by jstedfast** [comment]
Ping for reviews...

**Pull request merged**